### DISCUSSION TITLE
Pnp Search : search query on Active Directory property but not userProfile property ?

### DISCUSSION THREAD

**Discussion by DavidD410** [microsoft-search/pnp-modern-search#3590] [Q&A]
Hi,

in Pnp Search, I would like to bring up users, but only those who have a particular criteria in Active Directory (like company for example), a property which is not mapped with the user profile.
Is it possible to request directly from it ?
For example : _**companyName eq 'Microsoft - siège'**_

Thank you

**Comment by kasperbolarsen** [comment]
No, the people search data in SharePoint search comes from the User Profile Application so you will have to sync that field from the AAD to the UPA first.

**Comment by DavidD410** [comment]
However I can do it with the People Search spfx :

![2024-03-01_16h31_46](https://github.com/microsoft-search/pnp-modern-search/assets/160244835/089b1515-3373-41ec-aea1-75d32d81de1b)

**Comment by kasperbolarsen** [comment]
I am not familiar with that component. If it is querying the AAD then that would make sense. However SharePoint Search is using UPA.
You might see if Microsoft Search/Graph could be an option, as it seems like this dataset does contain CompanyName :-)
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/ac40e07b-916a-41c5-8c29-0f2898531779)

**Reply by DavidD410** [reply]
It seems to me that Pnp Search V4 uses Graph too, no ?
So it should also work with pnp search

**Reply by kasperbolarsen** [reply]
Microsoft search/Graph is a Data source in v4.x so yes. Graph Explorer will expose the data returned from the API

**Comment by patrikhellgren** [comment]
@DavidD410 That People Search web part doesn't use the search index at all like PnP Modern Search does. It uses regular Graph ODATA REST API calls, hence the web part settings for $select, $filter and $orderBy. So the name containing "Search" is a bit misleading.

**Reply by DavidD410** [reply]
Thank you for this information @patrikhellgren
